rosidl_runtime_c/include/rosidl_runtime_c/message_type_support_struct.h

```diff
@@ -28,21 +28,50 @@ typedef struct rosidl_message_type_support_t rosidl_message_type_support_t;
 typedef const rosidl_message_type_support_t * (* rosidl_message_typesupport_handle_function)(
   const rosidl_message_type_support_t *, const char *);
 
+/// Contains rosidl message type support data
 struct rosidl_message_type_support_t
 {
+  /// String identifier for the type_support.
   const char * typesupport_identifier;
+  /// Pointer to type support handle function
```

Comment:
I believe void * data is a library and func is the function pointer. Add comment to func. Address here and below.